src/test/java/wedlog/address/logic/parser/GuestAddCommandParserTest.java

```diff
@@ -1,10 +1,185 @@
+                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND;
+
+        assertThrows(ParseException.class, () -> parser.parse(NAME_DESC_AMY + validExpectedPersonString));
+    }
+
+    @Test
+    public void parse_invalidValues_failure() throws ParseException {
+        String validExpectedPersonString = NAME_DESC_BOB + PHONE_DESC_BOB
+                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND;
+
+        // invalid value followed by valid value
+
+        // invalid name
+        assertThrows(ParseException.class, () -> parser.parse(INVALID_NAME_DESC + validExpectedPersonString));
+
+        // invalid email
+        assertThrows(ParseException.class, () -> parser.parse(INVALID_EMAIL_DESC + validExpectedPersonString));
+
+
+        // valid value followed by invalid value
+
+        // invalid name
+        assertThrows(ParseException.class, () -> parser.parse(validExpectedPersonString + INVALID_NAME_DESC));
+
+        // invalid email
+        assertThrows(ParseException.class, () -> parser.parse(validExpectedPersonString + INVALID_EMAIL_DESC));
+
+        // invalid phone
+        assertThrows(ParseException.class, () -> parser.parse(validExpectedPersonString + INVALID_PHONE_DESC));
+
+        // invalid address
+        assertThrows(ParseException.class, () -> parser.parse(validExpectedPersonString + INVALID_ADDRESS_DESC));
+
+        // invalid rsvp
+        assertThrows(ParseException.class, () -> parser.parse(validExpectedPersonString + INVALID_RSVP_DESC));
+
+    }
+
+    @Test
+    public void parse_missingTags_success() {
+        // zero tags
+        Guest expectedGuest = new GuestBuilder(AMY).withTags().build();
+        assertParseSuccess(parser, NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY + ADDRESS_DESC_AMY
+                        + RSVP_DESC_AMY + DIETARY_DESC_AMY,
+                new GuestAddCommand(expectedGuest));
+    }
+
+    @Test
+    public void parse_missingPhone_success() {
+        Guest expectedGuest = new GuestBuilder(BOB).withoutPhone().build();
+        assertParseSuccess(parser, NAME_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+                + RSVP_DESC_BOB + DIETARY_DESC_BOB + TAG_DESC_FRIEND
+                + TAG_DESC_HUSBAND, new GuestAddCommand(expectedGuest));
+    }
+
+    @Test
+    public void parse_missingEmail_success() {
+
+        Guest expectedGuest = new GuestBuilder(BOB).withoutEmail().build();
+        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + ADDRESS_DESC_BOB
+                + RSVP_DESC_BOB + DIETARY_DESC_BOB + TAG_DESC_FRIEND
+                + TAG_DESC_HUSBAND, new GuestAddCommand(expectedGuest));
+    }
+
+    @Test
+    public void parse_missingAddress_success() {
+        Guest expectedGuest = new GuestBuilder(BOB).withoutAddress().build();
+        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
+                + RSVP_DESC_BOB + DIETARY_DESC_BOB + TAG_DESC_FRIEND
+                + TAG_DESC_HUSBAND, new GuestAddCommand(expectedGuest));
+
+    }
```

Comment:
Perhaps you could add similar tests `parse_missingRsvpStatus_success()` and `parse_missingDietaryRequirements_success()`?